Code/EnginePlugins/RecastPlugin/Components/DetourCrowdAgentComponent.cpp

```diff
@@ -0,0 +1,251 @@
+
+// clang-format off
+EZ_BEGIN_COMPONENT_TYPE(ezDetourCrowdAgentComponent, 1, ezComponentMode::Dynamic)
+{
+  EZ_BEGIN_ATTRIBUTES
+  {
+    new ezCategoryAttribute("AI/Recast"),
+  }
+  EZ_END_ATTRIBUTES;
+
+  EZ_BEGIN_PROPERTIES
+  {
+    EZ_MEMBER_PROPERTY("Radius",m_fRadius)->AddAttributes(new ezDefaultValueAttribute(0.3f)),
+    EZ_MEMBER_PROPERTY("Height",m_fHeight)->AddAttributes(new ezDefaultValueAttribute(1.8f)),
+    EZ_MEMBER_PROPERTY("MaxSpeed",m_fMaxSpeed)->AddAttributes(new ezDefaultValueAttribute(3.5f)),
+    EZ_MEMBER_PROPERTY("MaxAcceleration",m_fMaxAcceleration)->AddAttributes(new ezDefaultValueAttribute(10.0f)),
+    EZ_MEMBER_PROPERTY("StoppingDistance",m_fStoppingDistance)->AddAttributes(new ezDefaultValueAttribute(1.0f)),
+  }
+  EZ_END_PROPERTIES;
+}
+EZ_END_COMPONENT_TYPE
+// clang-format on
+
+ezDetourCrowdAgentComponent::ezDetourCrowdAgentComponent() = default;
+ezDetourCrowdAgentComponent::~ezDetourCrowdAgentComponent() = default;
+
+void ezDetourCrowdAgentComponent::SerializeComponent(ezWorldWriter& stream) const
+{
+  SUPER::SerializeComponent(stream);
+  ezStreamWriter& s = stream.GetStream();
+
+  s << m_fRadius;
+  s << m_fHeight;
+  s << m_fMaxSpeed;
+  s << m_fMaxAcceleration;
+  s << m_fStoppingDistance;
+}
+
+void ezDetourCrowdAgentComponent::DeserializeComponent(ezWorldReader& stream)
+{
+  SUPER::DeserializeComponent(stream);
+  ezStreamReader& s = stream.GetStream();
+
+  s >> m_fRadius;
+  s >> m_fHeight;
+  s >> m_fMaxSpeed;
+  s >> m_fMaxAcceleration;
+  s >> m_fStoppingDistance;
+}
+
+void ezDetourCrowdAgentComponent::FillAgentParams(ezDetourCrowdAgentParams& out_params) const
+{
+  out_params.m_fRadius = m_fRadius;
+  out_params.m_fHeight = m_fHeight;
+  out_params.m_fMaxSpeed = m_fMaxSpeed;
+  out_params.m_fMaxAcceleration = m_fMaxAcceleration;
+}
+
+void ezDetourCrowdAgentComponent::SetTargetPosition(const ezVec3& vPosition)
+{
+  m_vTargetPosition = vPosition;
+  m_PathToTargetState = ezAgentPathFindingState::HasTargetWaitingForPath;
+  m_bTargetDirty = true;
+}
+
+void ezDetourCrowdAgentComponent::ClearTargetPosition()
+{
+  m_PathToTargetState = ezAgentPathFindingState::HasNoTarget;
+}
+
+void ezDetourCrowdAgentComponent::OnActivated()
+{
+  SUPER::OnActivated();
+}
+
+void ezDetourCrowdAgentComponent::OnDeactivated()
+{
+  SUPER::OnDeactivated();
+}
+
+void ezDetourCrowdAgentComponent::OnSimulationStarted()
+{
+  SUPER::OnSimulationStarted();
+
+  ezCharacterControllerComponent* pCC = nullptr;
+  if (GetOwner()->TryGetComponentOfBaseType<ezCharacterControllerComponent>(pCC))
+  {
+    m_hCharacterController = pCC->GetHandle();
+  }
+}
+
+void ezDetourCrowdAgentComponent::SyncTransform(const ezVec3& vPosition, const ezVec3& vVelocity)
+{
+  m_vVelocity = m_vVelocity;
+
+  GetOwner()->SetGlobalPosition(vPosition);
+
+  //ezSimdTransform xform = GetOwner()->GetGlobalTransformSimd();
+  //xform.m_Position.Set(ezRcPos(pDtAgent->npos));
+  //xform.m_Rotation.
+  //GetOwner()->SetGlobalTransform(xform);
```

Comment:
You don't need to go through the SIMD functions, you can also just assign an ezTransform. Note that it is more efficient to call just one function (e.g. SetGlobalTransform) than two (e.g. SetGlobalPosition and then SetGlobalRotation).

Reply:
Thanks for the info!